darasimiolaifa/amistad
# chore(frontend): refactor state to use redux

## What does this PR do?
This PR refactors the application to use redux to manage state

## What major activities were carried out?
- install the redux packages
- create the redux scaffolds (store, types, actions, reducers)
- remove state from components and use the redux store for data

## How can this be manually tested?
- Clone the repository
- Fetch the branch
- Install the dependencies
- cd into the `amistad-client` folder
- Start project by running `npm start` in the terminal
- Your browser should automatically load to show you the homepage screen below:

![homepage-with-scream-cards](https://user-images.githubusercontent.com/28527393/71390146-6d5e3480-25ff-11ea-9526-7987d9b60d1f.png)
- In the browser, add `/signup` to the url. You should see the following page:

![signup page](https://user-images.githubusercontent.com/28527393/71532515-fd4bf900-28f3-11ea-9fdf-6a60ad58ada6.png)

- Fill in the form details as appropriate, then submit. The browser should redirect to take you to the homepage with `screams`.
- In another browser client. or in `incognito`, copy the current url and change `signup` in the url to `/login`. You should see the following page:

![Login page](https://user-images.githubusercontent.com/28527393/71513730-df52aa00-289b-11ea-80c1-80f469b12308.png)

- In the form, enter the email and password you just signed up with. The browser should redirect to take you to the homepage with `screams`.

## What background information should you know?
- Right now, the application stores the `token` from the backend in the browser's `localStorage` and unless that is deleted, any attempt to click either the `signup` or `login` buttons while logged in will redirect to the homepage. That's the reason you should swith to either another browser client or use the same browser in incognito mode to test the `login` route after signing up.
